### DISCUSSION TITLE
call firmware routines from Python code

### DISCUSSION THREAD

**Discussion by massimosala** [micropython/micropython#11885] [ESP8266]
Hi

Reading some of the firmware C files, I found interesting routines.

Like this adler32 implementation:
https://github.com/micropython/micropython/blob/master/lib/uzlib/adler32.c

Is it possible to call the routine from MicroPython... maybe using asm extensa?

In general, are the symbols of the routines published with their address? As standalone files or as information extractable with some tools from the firmware.bin ?

**Comment by rkompass** [comment]
It should be part of the MP module uzlib. I.e. you could use it like [`zlib.adler32()`](https://docs.python.org/3/library/zlib.html) in CPython.
Perhaps not in MP 1.19, perhaps not compiled in by some wrongly set option.
Ah, saw the source code - this is no source for a function available from within Micropython. You could look in the source tree, whether is is used somewhere. Perhaps it is planned to add such a MP function.
